docs/source/api/pystiche_papers.ulyanov_et_al_2016.rst

```diff
@@ -1,6 +1,103 @@
+  - :func:`~pystiche_papers.ulyanov_et_al_2016.style_loss`,
+  - :func:`~pystiche_papers.ulyanov_et_al_2016.transformer`.
+
+
+.. _table-branches-ulyanov_et_al_2016:
+
+The original authors have published two very similar papers, one with
+:class:`~torch.nn.BatchNorm2d` and a later one with :class:`~torch.nn.InstanceNorm2d`:
+
+  - ``master``: Corresponds to the reference implementation of the
+    `paper texture_nets_v2 <https://arxiv.org/abs/1701.02096>`_.
+  - ``texture_nets_v1``: Corresponds to the reference implementation of the
+    `paper texture_nets_v1 <https://arxiv.org/abs/1603.03417>`_.
+
+Unfortunately, the hyperparameters used differ from those in the papers, as well as
+among the individual implementations themselves. If you use ``instance_norm`` and
+``impl_params`` as in the following table, the appropriate parameters will be used:
+
+  - ``master``: The parameters specified in the implementation branch
+    `master <https://github.com/pmeier/texture_nets/tree/master>`_.
+  - ``texture_nets_v1``: The parameters specified in the implementation branch
+    `texture_nets_v1 <https://github.com/pmeier/texture_nets/tree/texture_nets_v1>`_.
+  - ``"paper"``: The parameters specified in the papers. Here ``instance_norm`` is only
+    used to use either :class:`~torch.nn.InstanceNorm2d` or
+    :class:`~torch.nn.BatchNorm2d`.
+
++-----------------+-----------+------------------------------------+
+|                 |           |        ``instance_norm``           |
++-----------------+-----------+-------------+----------------------+
+|                 |           |   ``True``  |       ``False``      |
++-----------------+-----------+-------------+----------------------+
+| ``impl_params`` | ``True``  | ``master``  | ``texture_nets_v1``  |
+|                 +-----------+-------------+----------------------+
+|                 | ``False`` | ``"paper"`` |      ``"paper"``     |
++-----------------+-----------+-------------+----------------------+
+
+.. _table-hyperparameters-ulyanov_et_al_2016:
+
+The following table provides an overview of the parameters:
+
+- the ``num_epochs`` of the :func:`~pystiche_papers.ulyanov_et_al_2016.training`,
+
+- the ``lr`` the learning rate of the
+  :func:`~pystiche_papers.ulyanov_et_al_2016.optimizer`,
+
+- the ``num_batches`` and ``batch_size`` of the
+  :func:`~pystiche_papers.ulyanov_et_al_2016.image_loader`,
+
+- the ``score_weight`` for
+
+  - the :func:`~pystiche_papers.ulyanov_et_al_2016.content_loss`,
+  - the :func:`~pystiche_papers.ulyanov_et_al_2016.style_loss`, as well as
+
+- the ``layers`` for
+
+  - the :func:`~pystiche_papers.ulyanov_et_al_2016.style_loss`.
+
+The ``layers`` have either ``4 layers`` ``("relu1_1", "relu2_1", "relu3_1", "relu4_1")``
+or ``5 Layer`` ``("relu1_1", "relu2_1", "relu3_1", "relu4_1", "relu5_1")``.
+
+
++-------------------+---------------+--------------+-----------------------+
+|   Parameter       |  ``"paper"``  |  ``master``  |  ``texture_nets_v1``  |
```

Comment:
Do we need the `paper` column? I think we should only list the reference implementation. At least that is what we did with [`johnson_alahi_li_2016`](https://github.com/pmeier/pystiche_papers/blob/master/docs/source/api/pystiche_papers.johnson_alahi_li_2016.rst).